src/document/UBDocumentProxy.cpp

```diff
@@ -134,6 +137,13 @@ void UBDocumentProxy::setWidgetCompatible(const QUuid &uuid, bool compatible)
     mWidgetCompatibility[uuid] = compatible;
 }
 
+bool UBDocumentProxy::needsCleanup()
+{
+    bool tmp = mNeedsCleanup;
+    mNeedsCleanup = false;
```

Comment:
It's quite odd to do it this way. Can you change it to the "classical" getter/setter method to avoid future confusion ? 

`if (myDoc->needsCleanup())` doesn't sound like you're modifying myDoc

Reply:
The idea was to make sure in the function that this returns `true` only exactly once. It's like a fuse that can only blow once. If I use classical getter/setter, then this logic would have to be at the caller side. I'm trying to find other examples for that, an probably a better name. E.g. the atomic functions use something like `fetchAndStore` to modify a value and return the previous one.

Reply:
still in this case I'm not sure it would add more benefit than it reduces readability. Maybe something like  `mCleanupPerformed` 

with, for example : 
```
if (!cleanupPerformed())
{
...
// unreferenced file or directory
                if (entry.isDir())
                {
                    qDebug() << "Deleting unreferenced directory" << filename;
                    if (!UBFileSystemUtils::deleteDir(absoluteFilePath))
                             qWarning() << "something went wrong during the cleanup process on directory" << filename;
                }
                else
                {
                    qDebug() << "Deleting unreferenced file" << filename;
                    if (!UBFileSystemUtils::deleteFile(absoluteFilePath))
                             qWarning() << "something went wrong during the cleanup process on document" << filename;
                }

...
mCleanupPerformed = true;
}
```

If you still think it is important and want to maintain it as-is, please provide a name that makes it really clear

Reply:
Indeed I would like to stick to the semantic, as **implementing** it is part of the `UBDocumentProxy`, while **using** this function is part of `UBPersistenceManager`. My proposal is to rename it to `testAndResetCleanupNeeded` to refer to the test-and-set semantics used for atomic variables. Here the atomic behavior is not important, as the operation occurs on the single event thread, but the operation is the same.